### PR TITLE
boards: bl654_dvk: Add BL654 DVK board

### PR DESCRIPTION
**Pull request by rerickson1**
Add the Laird Connectivity BL654 DVK board to zephyr.
This board is based off of the nrf52840-pca10056.

### CODE REVIEW

File: boards/arm/bl654_dvk/doc/bl654_dvk.rst [hunk 1]
```diff
@@ -0,0 +1,180 @@
+.. _bl654_dvk:
+
+BL654_DVK
+#################
+
+Overview
+********
+
+Zephyr applications use the bl654_dvk board configuration
```
Comment (dbkinder) [inline boards/arm/bl654_dvk/doc/bl654_dvk.rst]: We're defining some new intro paragraph guidelines that will let us generate a better board index page, so let's change this to:
```
The BL654 Development Kit hardware provides
support for the Laird BL654 module powered by a Nordic Semiconductor nRF52840 ARM Cortex-M4F CPU.

This development kit has the following features:
```

File: boards/arm/bl654_dvk/doc/bl654_dvk.rst [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+Segger IC.
+
+
+Testing the LEDs and buttons on the BL654 DVK
+************************************************
+
+There are 2 samples that allow you to test that the buttons (switches) and LEDs on
+the board are working properly with Zephyr:
+
+.. code-block:: console
```
Comment (dbkinder) [inline boards/arm/bl654_dvk/doc/bl654_dvk.rst]: delete this code-block and contents and replace with a list that links to the two sample's documentation:
```
* :ref:`blinky-sample`
* :ref:`button-sample`
```

File: boards/arm/bl654_dvk/doc/bl654_dvk.rst [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+   samples/basic/blinky
+   samples/basic/button
+
+You can build and flash the examples to make sure Zephyr is running correctly on
+your board. The button and LED definitions can be found in :file:
```
Comment (dbkinder) [inline boards/arm/bl654_dvk/doc/bl654_dvk.rst]: can't be a newline break here, so move the :file: to the next line:
```
:file:`boards/arm/bl654_dvk/bl654_dvk.dts`.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by aescolar** [comment]
Hi @rerickson1 : Please note that one test is failing in this new proposed board:
https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/34156/1/tests

**Comment by galak** [comment]
Looks good to me, I know @nashif made a few doc changes related to boards, so not sure if this is in sync with those.

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks.

**Comment by tautologyclub** [comment]
Is this without the external oscillator? If so, I wouldn't have expected all radio features to work. We had huge problems getting BLE to work with Zephyr on these boards before connecting the external osc.

**Comment by rerickson1** [comment]
> Is this without the external oscillator? If so, I wouldn't have expected all radio features to work. We had huge problems getting BLE to work with Zephyr on these boards before connecting the external osc.

By default the 32kHz ext oscillator is not connected.  We have one on the board that can be hooked up easily with some solder bridges.

**Review by anangl — APPROVED** [review]

**Comment by rerickson1** [comment]
Is there anything else needed on this.  Again, the shippable error seems unrelated to my changes.  Similar to the BL652 DVK board addition, I believe #13335 fixes the CI issue.

**Comment by galak** [comment]
Can you rebase to fix the merge conflict.

**Pull request merged**